### PR TITLE
Add tests for design.UserTypeExpr.IsCompatible()

### PR DESCRIPTION
**Pull request by tchssk**
#1423

### CODE REVIEW

File: design/user_type_test.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+		typ      DataType
+		values   []interface{}
+		expected bool
+	}{
+		"compatible": {
+			typ:      Int,
+			values:   []interface{}{i},
+			expected: true,
+		},
+		"not comatible": {
```
Comment (raphael) [inline design/user_type_test.go]: typo

Comment (tchssk) [reply design/user_type_test.go]: I fixed it. https://github.com/goadesign/goa/pull/1622/commits/a2809739d435deac4ab5b2b1cabd0db7bf2c421b

File: design/user_type_test.go [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			typ:      nil,
+			values:   []interface{}{b, i},
+			expected: true,
+		},
+	}
+
+	for k, tc := range cases {
+		u := UserTypeExpr{
+			AttributeExpr: &AttributeExpr{
+				Type: tc.typ,
```
[the remark this answers is not in the document]
Comment (tchssk) [reply design/user_type_test.go]: I changed the test data. https://github.com/goadesign/goa/pull/1622/commits/a2809739d435deac4ab5b2b1cabd0db7bf2c421b

**Comment by raphael** [comment]
And now I just introduced a case where user types can be non objects :P. This can happen for error types here: https://github.com/goadesign/goa/pull/1620/files#diff-a1d1dcd344d1570c1fde73169b76c43cR138 
So I guess this case should be tested as well, sorry!

**Comment by tchssk** [comment]
Oh I see. I remove https://github.com/goadesign/goa/pull/1622/commits/a2809739d435deac4ab5b2b1cabd0db7bf2c421b.

**Comment by tchssk** [comment]
Now this is ready to merge.

**Comment by raphael** [comment]
Thank you!

**Pull request merged**